src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -175,13 +208,33 @@ def _serialise_links(self) -> etree._Element:
         return prop_ele
 
 
+@dataclass
+class SegmentBounds:
+    segment_start: float | int | str
+    segment_end: float | int | str
```

Comment:
suggestion: IMO this should be correctly typed as `float`, and the checks, warning and conversion could be done either in the calling factory methods, a factory method on `SegmentBounds` or in the `__post_init__` method.  
I don't think there is any real reason to hold on to other types here, eventually they will be turned into float anyways..

Reply:
well it is not actually turned into a float is it? It is a string in the xml.
The reason why I allow it is because sometimes we read csv and force pandas to type everything datatype as string (except for empty). This is because pandas sometimes does weird things with dates. In those cases it is easier to specify that everything is a string. It is also possible that conversions may happen to floats that have a zero after the decimal point. It is really hard as a user to influence these datatype conversions. These are the reasons why I am open. In my opinion the most consistent would be to convert everything into a string and see if the regex matches... What do you think?

Reply:
Right. Well this is a question of handling invalid input. They dislike if we crash everything if the input is invalid. If we do a conversion of something that cannot be turned into a float we will crash... Actually an argument against storing it as a float is this problem:

```
i = np.nan
match i:
    case float():
        print("Found")
    case _:
        print("Not found")
```

The `np.nan` is recognised as float. `pd.NA` is not but there is no guarantee you will not encounter `np.nan` when using pandas. If we always converted it into a string and checked with a regex we would never crash and cover all bases. What do you think?